### PR TITLE
send messages from migration program updates

### PR DESCRIPTION
**Pull request by jaserud**
**Description of changes**

- Added array of programs being updated to migration object
- Messages for each program in array is sent after donors are migrated
- Added a spy in migration tests on the send function to verify it is called as expected

**Type of Change**

- [ ] Bug
- [ ] Refactor
- [x] New Feature

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [x] Manual testing
- [x] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)

### CODE REVIEW

File: src/submission/schema/schema-manager.ts [hunk 1]
```diff
@@ -361,7 +363,11 @@ namespace MigrationManager {
 
     const migrationAfterDonorCheck = await checkDonorDocuments(migration, newTargetSchema);
 
-    const migrationAfterSubmissionsCheck = await revalidateOpenSubmissionsWithNewSchema(
+    migrationAfterDonorCheck.programsWithDonorUpdates?.forEach(program =>
+      messenger.getInstace().sendProgramUpdatedMessage(program),
```
Comment (blabadi) [inline src/submission/schema/schema-manager.ts]: this doesn't throw errors right ?

Comment (jaserud) [reply src/submission/schema/schema-manager.ts]: Hmm the only one I can think of is if the instance is undefiend which it shouldn't be. All the asyn functions called inside of `sendMessage` are handled with `.catch`. But I guess better safe then sorry. so I could add catch logic here. Also my bad typo should be `getInstance`, will fix.

Comment (blabadi) [reply src/submission/schema/schema-manager.ts]: note that when you call for each for something that returns a promise, it will not await on it

Comment (jaserud) [reply src/submission/schema/schema-manager.ts]: I don't think we need to wait on the messages being sent though right? The only reason to wait would be if we wanted to know whether the message reached kafka, but I created a seperate ticket to handle errors if messages don't reach kafka.

Comment (blabadi) [reply src/submission/schema/schema-manager.ts]: I think so too, should be safe now after the try catch

**Pull request merged**